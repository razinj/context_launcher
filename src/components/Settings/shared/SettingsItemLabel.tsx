// React
import React from 'react'
// React Native
import { StyleSheet, Text, View } from 'react-native'
// Models
import { SettingsItemLabelProps as Props } from '../../../models/props'

const SettingsItemLabel = ({ title, description, titleStyle, wrapperStyle }: Props) => {
  return (
    <View style={[styles.wrapper, wrapperStyle]}>
      <Text style={[styles.label, titleStyle]}>{title}</Text>
      {description && <Text style={styles.description}>{description}</Text>}
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    padding: 10,
  },
  label: {
    color: '#808080',
  },
  description: {
    color: '#ccc',
  },
})

export default SettingsItemLabel
