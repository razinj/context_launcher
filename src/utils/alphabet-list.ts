import { AppDetails } from '../models/app-details'
import { AppLetterIndex } from '../models/list-letter-index'

type LetterMap = {
  [key: string]: number
}

const alphabet = [
  'A',
  'B',
  'C',
  'D',
  'E',
  'F',
  'G',
  'H',
  'I',
  'J',
  'K',
  'L',
  'M',
  'N',
  'O',
  'P',
  'Q',
  'R',
  'S',
  'T',
  'U',
  'V',
  'W',
  'X',
  'Y',
  'Z',
]

/**
 * @returns Indexed alphabet letters in an object where the key is the alphabet and the value is the index
 */
export const getLettersMap = (): LetterMap => {
  const letterMap: LetterMap = {}

  alphabet.forEach((letter: string, index: number) => (letterMap[letter] = index + 1))

  return letterMap
}

/**
 * @param value Any string value
 * @returns A letter if the first chareceter of the value is an alphabet, otherwise it returns a '#'
 */
export const getFirstLetter = (value: string): string => {
  const firstChar = value.trim().substring(0, 1).toUpperCase()
  const isValidLetter = getLettersMap()[firstChar]

  if (isValidLetter) return firstChar

  return '#'
}

export const getAppsLetterIndex = (apps: AppDetails[]): AppLetterIndex[] => {
  const appsLetterIndex: AppLetterIndex[] = []
  const treatedLetters: string[] = []

  apps.forEach(({ name }: AppDetails, index: number) => {
    const letter = getFirstLetter(name)

    if (treatedLetters.includes(letter)) return

    appsLetterIndex.push({ letter, index })
    treatedLetters.push(letter)
  })

  return appsLetterIndex
}
