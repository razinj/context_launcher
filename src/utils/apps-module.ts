// Native modules
import AppsModule from '../native-modules/AppsModule'

export const launchApp = (packageName: string) => AppsModule.launchApplication(packageName)

export const showAppDetails = (packageName: string) => AppsModule.showApplicationDetails(packageName)

export const requestAppUninstall = (packageName: string) => AppsModule.requestApplicationUninstall(packageName)
