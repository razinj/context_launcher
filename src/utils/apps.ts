// Models
import { AppDetails, AppDetailsWithIcon } from '../models/app-details'
import { PinnedApp } from '../models/pinned-app'

export const getAppsByLabel = (apps: AppDetails[], query: string): AppDetails[] => {
  return apps.filter((app: AppDetails) => app.label.match(new RegExp(query, 'gi')))
}

export const getPinnedAppByName = (pinnedApps: PinnedApp[], appDetails: AppDetailsWithIcon): PinnedApp => {
  const pinnedApp = pinnedApps.find(({ name }: PinnedApp) => name === appDetails.name)

  if (pinnedApp) return pinnedApp

  return { ...appDetails, isTemporary: false, isPermanent: false }
}
