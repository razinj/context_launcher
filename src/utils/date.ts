export const getTimeFromDate = (stringDate?: string): string => {
  if (!stringDate) return 'unset'

  const date = new Date(stringDate)

  const singleDigitHours = `${date.getHours()}`.length == 1
  const singleDigitMinutes = `${date.getMinutes()}`.length == 1

  const hours = singleDigitHours ? `0${date.getHours()}` : date.getHours()
  const minutes = singleDigitMinutes ? `0${date.getMinutes()}` : date.getMinutes()

  return `${hours}:${minutes}`
}

export const getDateFromStringWithCurrentDateValue = (stringDate?: string): Date | undefined => {
  if (!stringDate) return

  const currentDate = new Date()
  const fullDate = new Date(stringDate)
  return new Date(
    currentDate.getFullYear(),
    currentDate.getMonth(),
    currentDate.getDate(),
    fullDate.getHours(),
    fullDate.getMinutes(),
    0,
    0
  )
}

export const stripDateFromSeconds = (date: Date): Date => {
  return new Date(date.getFullYear(), date.getMonth(), date.getDate(), date.getHours(), date.getMinutes(), 0, 0)
}
