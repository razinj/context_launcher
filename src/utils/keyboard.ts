import { Keyboard } from 'react-native'

export const dismissKeyboard = () => Keyboard.dismiss()
