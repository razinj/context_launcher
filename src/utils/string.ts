export const truncateString = (value: string | undefined, maxLength: number): string | undefined => {
  return value && value.length > maxLength ? `${value.substring(0, maxLength)}...` : value
}
