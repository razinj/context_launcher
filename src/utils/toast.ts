import { ToastAndroid } from 'react-native'

export const displayToast = (message: string, duration: number = ToastAndroid.SHORT) => {
  ToastAndroid.show(message, duration)
}
