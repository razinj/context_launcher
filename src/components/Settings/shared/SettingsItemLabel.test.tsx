import { screen } from '@testing-library/react-native'
import React from 'react'
import { renderWithProvider } from '../../../../utils/test/utils'
import SettingsItemLabel, { SettingsItemLabelProps } from '../shared/SettingsItemLabel'

describe('<SettingsItemLabel /> Tests', () => {
  const requiredProps: SettingsItemLabelProps = {
    title: 'An item label',
  }

  it('should render correctly and match snapshot - required props', () => {
    renderWithProvider(<SettingsItemLabel {...requiredProps} />)

    expect(screen.toJSON()).toMatchSnapshot()
    expect(screen.getByText(requiredProps.title)).toBeOnTheScreen()
  })

  it('should render correctly and match snapshot - required and optional props', () => {
    const allProps: SettingsItemLabelProps = {
      ...requiredProps,
      description: 'An item description',
      titleStyle: { color: '#000' },
      wrapperStyle: { flex: 1 },
    }

    renderWithProvider(<SettingsItemLabel {...allProps} />)

    expect(screen.toJSON()).toMatchSnapshot()
    expect(screen.getByText(allProps.title)).toBeOnTheScreen()
    expect(screen.getByText(allProps.description!)).toBeOnTheScreen()
    expect(screen.getByTestId('wrapper')).toHaveStyle(allProps.wrapperStyle)
    expect(screen.getByTestId('title')).toHaveStyle(allProps.titleStyle)
  })
})
