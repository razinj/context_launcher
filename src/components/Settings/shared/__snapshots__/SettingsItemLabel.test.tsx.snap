// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsItemLabel /> Tests should render correctly and match snapshot - required and optional props 1`] = `
<View
  style={
    [
      {
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#808080",
        },
        undefined,
      ]
    }
  >
    An item label
  </Text>
  <Text
    style={
      {
        "color": "#ccc",
      }
    }
  >
    An item description
  </Text>
</View>
`;

exports[`<SettingsItemLabel /> Tests should render correctly and match snapshot - required props 1`] = `
<View
  style={
    [
      {
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#808080",
        },
        undefined,
      ]
    }
  >
    An item label
  </Text>
</View>
`;
